libsel4vm/src/arch/x86/processor/decode.c

```diff
@@ -368,15 +499,65 @@ uintptr_t vm_emulate_realmode(vm_vcpu_t *vcpu, uint8_t *instr_buf,
             case 0xa1:
                 /* mov offset memory to eax */
                 instr++;
+#ifdef CONFIG_ARCH_X86_64
+                memcpy(&mem, instr, 4);
```

Comment:
When using `sizeof(seL4_Word)/2` there is no need to duplicate the code. Might apply in other places also.
```suggestion
                memcpy(&mem, instr, sizeof(seL4_Word)/2 );
```